src/Workspaces/CSharp/Portable/EmbeddedLanguages/VirtualChars/CSharpVirtualCharService.cs

```diff
@@ -135,6 +136,20 @@ private VirtualCharSequence TryConvertVerbatimStringToVirtualChars(SyntaxToken t
 
                     index += result.Last().Span.Length;
                 }
+                else if (char.IsHighSurrogate(tokenText[index]))
+                {
+                    // we had a high surrogate.  we better have a following low surrogate for this to be a legal string.
+                    if (index + 1 >= endIndexExclusive || !char.IsLowSurrogate(tokenText[index + 1]))
+                    {
+                        Debug.Fail("This should not be reachable as long as the compiler added no diagnostics.");
```

Comment:
>Debug.Fail("This should not be reachable as long as the compiler added no diagnostics."); [](start = 24, length = 89)

`Contract.ThrowIfFalse(index + 1 >= endIndexExclusive || !char.IsLowSurrogate(tokenText[index + 1]))`

Reply:
i don't want to throw here.  The pattern of the VCService is that it makes some assumptions about how the underlying system works, but if those assumptions are wrong, it just gracefully bails instead of throwing.  The idea here is that hte assumptions aren't documented/guaranteed.  They come from my careful reading of things, but could be incorrect.  I don't want to throw on invariants that may not exist.  (note: this is the pattern for the rest of hte type).

Reply:
I see. Ok.